src/Payments/Api.php

```diff
@@ -80,8 +80,12 @@ public function add_payment_method_script_dependencies( $dependencies, $handle )
 	 * Add payment method data to Asset Registry.
 	 */
 	public function add_payment_method_script_data() {
-		$script_data = $this->payment_method_registry->get_all_registered_script_data();
+		// Enqueue the order of enabled gateways as `paymentGatewaySortOrder`.
+		$available_gateways = WC()->payment_gateways->get_available_payment_gateways();
```

Comment:
> Note - COD is not always displayed in merchant-configured order

Could that be fixed if replacing `get_available_payment_gateways()` with `payment_gateways()`? The latter returns all payment methods, not only the available ones, so COD would be in the array even if not available when loading the block.

Reply:
Good thinking! I've added that here 88b35a4cb85b108680915937eab3284a68e2ac20 and it seems to work well. 

One risk with this is that it adds every gateway to the list, even if they are disabled by the merchant:

<img width="1837" alt="Screen Shot 2020-08-05 at 11 04 31 AM" src="https://user-images.githubusercontent.com/4167300/89353851-74a7fd80-d70b-11ea-842d-aadb5c8e4900.png">

<img width="466" alt="Screen Shot 2020-08-05 at 11 03 31 AM" src="https://user-images.githubusercontent.com/4167300/89353810-5e01a680-d70b-11ea-8a02-2160aa74fcbd.png">

That's probably fine, just a little inefficient. I can't imagine a merchant would want to protect this information (what gateway extensions are installed) but if so this could be considered a leak :)

Reply:
Good observation, but I don't think this is an information leak. What actionable things can be done with this that would be harmful to merchants?

Reply:
Off hand I can't think of any :)